### PR TITLE
Refactor Optimize function

### PR DESCRIPTION
**Pull request by aykevl**
This refactor is a prerequisite to a much larger refactor in the compiler, see #746.

### CODE REVIEW

**Comment by aykevl** [comment]
Rebased to make CI succeed.

**Comment by deadprogram** [comment]
Looks good to me. @jaddr2line any feedback before I merge this?

File: transform/coroutines.go [hunk 1]
```diff
@@ -5,11 +5,27 @@ package transform
...
+	"internal/task.Pause",
+	"internal/task.fake",
+	"internal/task.Current",
+	"internal/task.createTask",
+	"(*internal/task.Task).setState",
+	"(*internal/task.Task).returnTo",
+	"(*internal/task.Task).returnCurrent",
+	"(*internal/task.Task).setReturnPtr",
+	"(*internal/task.Task).getReturnPtr",
+}
```
Comment (jaddr2line) [inline transform/coroutines.go]: These should be put near their use - maybe right after `functionsUsedInTransforms`.

File: transform/optimizer.go [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-	"github.com/tinygo-org/tinygo/transform"
+	"github.com/tinygo-org/tinygo/compileopts"
+	"github.com/tinygo-org/tinygo/compiler/ircheck"
 	"tinygo.org/x/go-llvm"
 )
 
 // Run the LLVM optimizer over the module.
 // The inliner can be disabled (if necessary) by passing 0 to the inlinerThreshold.
-func (c *Compiler) Optimize(optLevel, sizeLevel int, inlinerThreshold uint) []error {
+func Optimize(mod llvm.Module, config *compileopts.Config, optLevel, sizeLevel int, inlinerThreshold uint) []error {
```
Comment (jaddr2line) [inline transform/optimizer.go]: This function does a lot more than just run the LLVM optimizer. It might make sense to update the comment.

**Comment by deadprogram** [comment]
Thanks @aykevl and @jaddr2line now merging.

File: transform/optimizer.go (hunk 2 above)
Comment (aykevl) [reply transform/optimizer.go]: Oops forgot about this one. I did update the comment in a previous attempt but forgot it this time.

**Pull request merged**